extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -93,112 +110,409 @@ public function addPage( $id, $data ) {
-					"last_revision_id" => $row->last_rev_id,
-					"comment_timestamp" => $row->last_touched,
-					"display_order" => $display_order ++,
-				] );
-			} );
+			unset( $inserts );
+		}
 
 		return $this->pages;
 	}
 
-	public function getRevisions() {
-		if ( !empty( $this->revisions ) ) {
-			return $this->revisions;
+	/**
+	 * Revision
+	 * +----------------+---------------------+------+-----+----------------+----------------+
+	 * | Field          | Type                | Null | Key | Default        | Extra          |
+	 * +----------------+---------------------+------+-----+----------------+----------------+
+	 * | rev_id         | int(10) unsigned    | NO   | PRI | NULL           | auto_increment |
+	 * | rev_page       | int(10) unsigned    | NO   | MUL | NULL           |                |
+	 * | rev_text_id    | int(10) unsigned    | NO   |     | NULL           |                |
+	 * | rev_comment    | tinyblob            | NO   |     | NULL           |                |
+	 * | rev_user       | int(10) unsigned    | NO   | MUL | 0              |                |
+	 * | rev_user_text  | varchar(255)        | NO   | MUL |                |                |
+	 * | rev_timestamp  | binary(14)          | NO   | MUL |                |                |
+	 * | rev_minor_edit | tinyint(3) unsigned | NO   |     | 0              |                |
+	 * | rev_deleted    | tinyint(3) unsigned | NO   |     | 0              |                |
+	 * | rev_len        | int(10) unsigned    | YES  |     | NULL           |                |
+	 * | rev_parent_id  | int(10) unsigned    | YES  |     | NULL           |                |
+	 * | rev_sha1       | varbinary(32)       | NO   |     |                |                |
+	 * +----------------+---------------------+------+-----+----------------+----------------+
+	 *
+	 * Text
+	 * +-----------+------------------+------+-----+---------+----------------+
+	 * | Field     | Type             | Null | Key | Default | Extra          |
+	 * +-----------+------------------+------+-----+---------+----------------+
+	 * | old_id    | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
+	 * | old_text  | mediumblob       | NO   |     | NULL    |                |
+	 * | old_flags | tinyblob         | NO   |     | NULL    |                |
+	 * +-----------+------------------+------+-----+---------+----------------+
+	 */
+	public function getRevisions( $fh ) {
+		$revIds = [];
+
+		foreach ( $this->getPages() as $id => $data ) {
+			$revIds[] = $data['latest_revision_id'];
 		}
 
+		$chunks = array_chunk($revIds, 500);
+		$chunksNumber = count( $chunks );
+
+		$currentBatch = 1;
+
+		foreach ($chunks as $part) {
+
+			WikiaLogger::instance()->info( "Batch " . $currentBatch . " of " . $chunksNumber );
+			$currentBatch++;
+
+			$tries = 3;
+			$queryResult = null;
+
+			do {
+				$dbh = DumpUtils::getDBSafe( DB_SLAVE );
+				$dbh->ping();
+				$inserts = [];
+				( new \WikiaSQL() )->SELECT( "revision.*, text.*" )
```

Comment:
again, do we really need all fields from these tables?

Reply:
Maybe we could get rid of few of them, but I don't want to break something as later I am building Revision and Title objects from this row.